cpp/src/decisiontree/levelalgo/levelkernel_classifier.cuh

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include "common_kernel.cuh"
+#include "cub/cub.cuh"
+
+__global__ void gini_kernel_level(const int* __restrict__ labels,
```

Comment:
knitpicky question: why do you call this `gini_kernel_level` when this kernel seems to compute histograms?

Reply:
yeah. just a name thing. this kernel is used just once. any suggestions for change of name?

Reply:
maybe `sample_count_histogram_kernel`?

Reply:
Done. Thanks